app/controllers/logix_controller.rb

```diff
@@ -2,14 +2,17 @@ class LogixController < ApplicationController
   # before_action :authenticate_user!
 
   def index
-      @projects = Project.select("id,author_id,image_preview,name").where(project_access_type:"Public",forked_project_id:nil).paginate(:page => params[:page]).order("id desc").limit(Project.per_page)
-
-      respond_to do |format|
-        format.html # index.html.erb
-        format.json { render json: @projects }
-        format.js
-      end
+    @projects = Project.select("id,author_id,image_preview,name")
+                       .where(project_access_type: "Public", forked_project_id: nil)
+                       .paginate(page: params[:page]).order("id desc").limit(Project.per_page)
+
+    respond_to do |format|
+      format.html # index.html.erb
+      format.json { render json: @projects }
+      format.js
+    end
 
+    @featured_circuits = FeaturedCircuit.includes(:project).take(4)
```

Comment:
use limit here so that Database will handle limiting records, also don't use magic numbers in the code 

https://www.eliotsykes.com/magic-numbers

Reply:
Working on this one.